opsdroid/connector/matrix/connector.py

```diff
@@ -193,6 +193,8 @@ async def connect(self):
             encryption_enabled=self._allow_encryption,
             pickle_key="",
             store_name="opsdroid.db" if self._allow_encryption else "",
+            request_timeout=60,
+            max_timeouts=1,
```

Comment:
I am not sure we want to have `max_timeouts` as 1. Quite often opsdroid is running somewhere out of sight, having it try to login once, throw an error and give up seems... unhelpful.

I think the main issue with the original issue I opened, was the lack of feedback about login failing. As long as we log what's happening I think we could keep the default number of retries as infinitiy.

Reply:
We can set an higher value for max_timeouts but we don't have any feedback about the number of attempts.
We have to wait for max_timeouts to be reached before we can log anything.
An other approach could be to set max_timeouts = 1 and manually retry when the exception is raised. 
But I 'm not sure that this is the way Opsdroid usually works.

Reply:
> We have to wait for max_timeouts to be reached before we can log anything.

ah, ok that's interesting.

> An other approach could be to set max_timeouts = 1 and manually retry when the exception is raised.

This feels like a good answer, but I agree, I am not sure how the other connectors handle connection errors. It seems like [slack just errors](https://github.com/opsdroid/opsdroid/blob/master/opsdroid/connector/slack/connector.py#L115) and the core has no retry functionality. So perhaps we should just follow them and keep this as it is.